daisy_workflows/image_import/windows/run_startup_scripts.cmd

```diff
@@ -18,13 +18,25 @@ ping 127.0.0.1 -n 60
 
 echo "Translate: Starting image translate..." > COM1:
 
-REM Restart the system in 5 minutes. Translate.ps1 will cancel this restart when it starts.
-REM This is to address initial boot issues, mostly on 2008R2, where the network interface is not yet usable.
-echo "Scheduling restart in 5 minutes. Translate.ps1 will cancel this restart." > COM1:
-shutdown /r /t 300
+for /f "tokens=4-5 delims=. " %%i in ('ver') do set version_number=%%i.%%j
+echo "Windows Version Number: %version_number%" > COM1:
 
-echo "Running network.ps1 to reconfigure network to DHCP if needed and log DNS and connectivity tests." > COM1:
-PowerShell.exe -NoProfile -NoLogo -ExecutionPolicy Unrestricted -File "%ProgramFiles%\Google\Compute Engine\metadata_scripts\network.ps1" > COM1:
+set has_network_interface_api=1
+if %version_number% == "6.0" ( set has_network_interface_api=0 )
+if %version_number% == "6.1" ( set has_network_interface_api=0 )
+
+if %has_network_interface_api% EQU 0 (
+  REM Restart the system in 10 minutes. Translate.ps1 will cancel this restart when it starts.
+  REM This is to address initial boot issues, mostly on 2008R2, where the network interface is not yet usable.
+
+  echo "Scheduling restart in 10 minutes. Translate.ps1 will cancel this restart." > COM1:
+  shutdown /r /t 600
+)
+
+if %has_network_interface_api% EQU 1 (
```

Comment:
Instead of not running this on older OSes, you could add the Windows 7/2008 versions of the cmdlets so you have this critical troubleshooting information for older OSes.